controllers/datafoundationclient.go

```diff
@@ -248,3 +275,44 @@ func (r *dataFoundationClientReconciler) reconcileCSIKMSConnectionDetailsConfigM
 
 	return nil
 }
+
+func (r *dataFoundationClientReconciler) hasOCSVolumes() (bool, error) {
+	storageClassList := &storagev1.StorageClassList{}
+	if err := r.list(storageClassList); err != nil {
+		return false, fmt.Errorf("failed to list Storage Classes: %v", err)
+	}
+	ocsStorageClass := map[string]bool{}
+	for i := range storageClassList.Items {
+		storageClass := storageClassList.Items[i]
+		if storageClass.Provisioner == r.offering.Namespace+ocsCephFsProvisionerSuffix || storageClass.Provisioner == r.offering.Namespace+ocsRBDProvisionerSuffix {
+			ocsStorageClass[storageClass.Name] = true
+		}
+	}
```

Comment:
Why do we need to calculate the provisioners name on each loop iteration?
```suggestion
	ocsStorageClass := map[string]bool{}
	rbdProvisionerName = r.offering.Namespace + ocsRBDProvisionerSuffix
	fsProvisionerName := r.offering.Namespace + ocsCephFsProvisionerSuffix
	
	for i := range storageClassList.Items {
		storageClass := storageClassList.Items[i]
		if storageClass.Provisioner == rbdProvisionerName || storageClass.Provisioner == fsProvisionerName {
			ocsStorageClass[storageClass.Name] = true
		}
	}
```

Reply:
updated